tests/cypress/integration/z-newfold-labs/wp-module-onboarding/2-intermediate-steps/1-top-priority/TC_07-top-priority.cy.js

```diff
@@ -0,0 +1,81 @@
+// <reference types="Cypress" />
+import { DrawerActivityForMenu } from '../../wp-module-support/drawer.cy';
+import { CheckHeadingSubheading } from '../../wp-module-support/header.cy';
+
+describe( 'Top Priority Page', function () {
+	before( () => {
+		cy.visit( 'wp-admin/?page=nfd-onboarding#/wp-setup/step/top-priority' );
+	} );
+
+	it( 'Check Drawer Activity', () => {
+		DrawerActivityForMenu(
+			'Exit to WordPress',
+			':nth-child(2)',
+			'Top Priority'
+		);
+	} );
+
+	it( 'Check if Header and Subheader shows up', () => {
+		CheckHeadingSubheading();
+	} );
+
+	it( 'Check if different cards are present', () => {
+		cy.get( '.selectable_cards' ).should( 'be.visible' );
+	} );
+
+	it( 'Check existence of Skip this Step button', () => {
+		cy.get( '.components-button.skip-button' ).should( 'be.visible' );
+	} );
+
+	it( 'Click on different cards and move on to next page', () => {
+		let previewCount = 0;
+		const IndexCommerce = 0;
+		const className = '.components-surface.components-card.nfd-card';
+		const arr = cy.get( className );
+		arr.each( () => {
+			cy.get( className )
+				.eq( previewCount )
+				.should( 'be.visible' )
+				.click();
+			previewCount += 1;
+			cy.get( '.nfd-card__body.nfd-selected-card' )
+				.find( '.nfd-card__body_title' )
+				.invoke( 'text' )
+				.then( ( $ele ) => {
+					if ( $ele !== 'Selling' ) {
+						cy.log( previewCount );
+						cy.get(
+							'.components-button.navigation-buttons.navigation-buttons_next.is-primary'
+						).click();
+						cy.url().should( 'include', 'step/basic-info', {
+							timeout: 30000,
+						} );
+						cy.go( 'back' );
+						cy.get(
+							'.nfd-card__body.nfd-selected-card> .nfd-card__body_title'
+						).contains( $ele );
+					}
+				} );
+		} );
+		cy.get( className )
+			.eq( 1 )
+			.then( ( $el ) => {
+				cy.get( className ).eq( 1 ).should( 'be.visible' ).click();
+				cy.get( '.nfd-card__body.nfd-selected-card' )
+					.find( '.nfd-card__body_title' )
+					.invoke( 'text' )
+					.then( ( $ele ) => {
+						cy.exec(
+							'npx wp-env run cli wp plugin activate woocommerce'
```

Comment:
Should we deactivate WooCommerce again in an `after` so that other test cases are not affected?

Reply:
None of the test cases are being affected since we have ecommerce steps after this page.

Reply:
They're not being affected for now because we have the order figured out, we are also trying to assert two flows at the same time, sometimes passing `flow=ecommerce` and sometimes not, this can become hard to debug in the future as we start adding more flow specific assertions, we need to have some kind of flow separation and also think about not ordering all our test cases (user can always jump back and forth using the drawer).

A very, very simple development issue that I can immediately think of caused by ordering them is:
Let's say I have to add an intermediate-step after `basic-info` named `theme-selection`, in the current scenario I would have to name the directory `4-theme-selection`, and because of the way this is structured, I have to rename all the other ones and increment them `4-design => 5-design` so on........

Let's think about how we can have separation of flows and keep ordering to a minimum? This will also allow us to isolate these tests and make writing them easier in the future.